pyroma/ratings.py

```diff
@@ -266,13 +268,39 @@ def test(self, data):
 
     def message(self):
         if self._major_version_specified:
-            return (
-                "The classifiers should specify what minor versions of "
-                "Python you support as well as what major version."
-            )
+            return "The classifiers should specify what minor versions of "\
+                   "Python you support as well as what major version."
         return "You should specify what Python versions you support."
 
 
+class PythonRequiresVersion(BaseTest):
+    def test(self, data):
+        self.weight = 100
+        python_requires = data.get("python_requires", None)
+        requires_python = data.get("requires_python", None)
+
+        # python_requires and requires_python are turned into 'Requires-Python' header
+        # https://packaging.python.org/en/latest/guides/dropping-older-python-versions/#specify-the-version-ranges-for-supported-python-distributions
+        # Some tools seems to report only one (eg: test_complete, test_markdown fail when 'requires_python' is not checked).
+        # So using whichever has a value provided
+        python_requires = python_requires or requires_python
+
+        if not python_requires:
+            return False
+
+        try:
+            from packaging.specifiers import SpecifierSet
+            SpecifierSet(python_requires)
+            return True
+        except packaging.specifiers.InvalidSpecifier as e:
+            return False
+
+    def message(self):
+        return (
+            "You should specify what Python versions you support with 'python_requires=....'."
```

Comment:
This is not only Setuptools- but `setup.py`-only syntax, which is now heavily discouraged for most modern projects. Therefore, we should say something like

```suggestion
            "You should specify what Python versions you support with "
            "the 'requires-python'/'python_requires' metadata."
```